### PR TITLE
JBPM-7266 Include Narayana + DBCP integration into persistence tests setup

### PR DESCRIPTION
**Pull request by rsynek**
Introduces tomcat-dbcp pooling data source into persistence tests setup.

### CODE REVIEW

**Comment by baldimir** [comment]
jenkins retest this

**Comment by rsynek** [comment]
jenkins retest this

**Review by baldimir — APPROVED** [review]

**Comment by rsynek** [comment]
@mariofusco could you please take a look? There are another PRs in jBPM depending on this one. Thanks!

**Comment by rsynek** [comment]
execute full downstream build

**Comment by rsynek** [comment]
jenkins execute full downstream build

**Comment by mariofusco** [comment]
@rsynek This change seems to have broken some persistence related tests in optaplanner. Please check them before I merge this pull request.

**Comment by rsynek** [comment]
@mariofusco yes - very likely the same changes with DBCP pooling datasource will be required for OptaPlanner persistence too. I will hopefully get to it today.

**Comment by rsynek** [comment]
jenkins execute full downstream build

**Comment by rsynek** [comment]
Fixed the corresponding TransactionManager changes in jBPM.

**Comment by rsynek** [comment]
jenkins execute full downstream build

**Comment by rsynek** [comment]
Recent failures are not connected to this PR. Will retry the FDB once it's resolved.

**Comment by rsynek** [comment]
jenkins execute full downstream build

**Comment by rsynek** [comment]
PR check on jBPM and this FDB show failures in modules that were not covered by preliminary testing. Merging will have to wait until these are resolved.

**Comment by rsynek** [comment]
jenkins retest this

**Comment by rsynek** [comment]
jenkins execute full downstream build

**Comment by rsynek** [comment]
jenkins execute full downstream build

**Comment by rsynek** [comment]
jenkins retest this

**Comment by rsynek** [comment]
@baldimir @mariofusco the PR has been changed significantly; could you please review once more? The aim is to reuse the code - except for PersistenceUtil - in jBPM too.

File: kie-test-util/src/main/java/org/kie/test/util/db/DataSourceFactory.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+import java.util.Properties;
+
+import org.apache.tomcat.dbcp.dbcp2.managed.BasicManagedDataSource;
+import org.kie.test.util.db.internal.DatabaseProvider;
+import org.kie.test.util.db.internal.PoolingDataSourceWrapper;
+
+/**
+ * Creates instances of {@link PoolingDataSourceWrapper} that can be used for testing purposes.
+ */
+final public class DataSourceFactory {
```
Comment (baldimir) [inline kie-test-util/src/main/java/org/kie/test/util/db/DataSourceFactory.java]: Nitpicking: public final

File: kie-test-util/src/main/java/org/kie/test/util/db/DataSourceFactory.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+                for (String propertyName : new String[]{"serverName", "portNumber", "databaseName"}) {
+                    sanitizedDriverProperties.put(propertyName, driverProperties.getProperty(propertyName));
+                }
+                sanitizedDriverProperties.put("URL", driverProperties.getProperty("url"));
+            } else if (databaseProvider == DatabaseProvider.MYSQL || databaseProvider == DatabaseProvider.MARIADB) {
+                for (String propertyName : new String[]{"databaseName", "serverName", "portNumber", "url"}) {
+                    sanitizedDriverProperties.put(propertyName, driverProperties.getProperty(propertyName));
+                }
+            } else if (databaseProvider == DatabaseProvider.SYBASE) {
+                for (String propertyName : new String[]{"databaseName", "portNumber", "serverName", "url"}) {
```
Comment (baldimir) [inline kie-test-util/src/main/java/org/kie/test/util/db/DataSourceFactory.java]: You can join this with MYSQL and MARIADB.. they are the same.

File: kie-test-util/src/main/java/org/kie/test/util/db/DataSourceFactory.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+                    sanitizedDriverProperties.put(propertyName, driverProperties.getProperty(propertyName));
+                }
+                // com.edb is Postgres Plus.
+            } else if (databaseProvider == DatabaseProvider.POSTGRES || databaseProvider == DatabaseProvider.POSTGRES_PLUS) {
+                for (String propertyName : new String[]{"databaseName", "portNumber", "serverName", "url"}) {
```
Comment (baldimir) [inline kie-test-util/src/main/java/org/kie/test/util/db/DataSourceFactory.java:89]: You can join this with MYSQL and MARIADB.. they are the same.

File: kie-test-util/src/main/java/org/kie/test/util/db/DataSourceFactory.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+                sanitizedDriverProperties.put("databaseName", driverProperties.getProperty("databaseName"));
+                sanitizedDriverProperties.put("driverType", "4");
+                sanitizedDriverProperties.put("serverName", driverProperties.getProperty("serverName"));
+                sanitizedDriverProperties.put("portNumber", driverProperties.getProperty("portNumber"));
+                sanitizedDriverProperties.put("currentSchema", driverProperties.getProperty("defaultSchema"));
+                sanitizedDriverProperties.put("url", driverProperties.getProperty("url"));
+                sanitizedDriverProperties.put("ResultSetHoldability", "1");
+                sanitizedDriverProperties.put("DowngradeHoldCursorsUnderXa", "true");
+            } else if (databaseProvider == DatabaseProvider.MSSQL) {
+                for (String propertyName : new String[]{"serverName", "portNumber", "databaseName"}) {
```
Comment (baldimir) [inline kie-test-util/src/main/java/org/kie/test/util/db/DataSourceFactory.java:80]: You could do this like you will fill serverName, portNumber, databaseName for all cases and then add url for the remaining ones.

**Review by baldimir — CHANGES_REQUESTED** [review]
Other than the comments, please check "final" keyword usage consistency.

**Comment by rsynek** [comment]
@baldimir thanks for the review, fixed (changed the DB checking a little bit differently, though).

**Review by baldimir — APPROVED** [review]
Approving if checks are green.

**Comment by rsynek** [comment]
@baldimir there might be a PR to OptaPlanner required before we run FDB, will check.

**Comment by rsynek** [comment]
jenkins execute full downstream build

**Comment by rsynek** [comment]
@baldimir fixed the setup for OptaPlanner, running the FDB, let's see.

**Comment by rsynek** [comment]
jenkins execute full downstream build

**Comment by rsynek** [comment]
jenkins execute full downstream build

**Review by mswiderski — APPROVED** [review]

**Comment by rsynek** [comment]
FDN build failed due to enforcer rules; already fixed in a related PR to jBPM.

**Comment by rsynek** [comment]
jenkins execute full downstream build

**Pull request merged**